File: backend/HskLearn.Api/Models/HskWord.cs
namespace HskLearn.Api.Models;

public record HskWord(
    int Id,
    string Word,
    string Pinyin,
    string PartOfSpeech,
    string Translation,
    int HskLevel = 4,
    string? ExampleSentence = null,
    string? ExamplePinyin = null,
    string? ExampleTranslation = null
);

File: android/app/src/main/java/com/hsklearn/app/data/model/HskWord.kt
package com.hsklearn.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class HskWord(
    val id: Int,
    val word: String,
    val pinyin: String,
    val partOfSpeech: String,
    val translation: String,
    val hskLevel: Int = 4,
    val exampleSentence: String? = null,
    val examplePinyin: String? = null,
    val exampleTranslation: String? = null,
)
